### PR TITLE
Update README and site with skydoc deprecation

### LINKED ISSUE #146: README for the repo is stale
it's still referring to the skydoc docs, not the stardoc ones (https://skydoc.bazel.build/docs/getting_started_stardoc.html) and generally refers to everything as skydoc instead of stardoc.  It could use a refresh.

### PR DESCRIPTION
**Pull request by c-parsons**
* Update README to reference Stardoc instead of Skydoc
* Improve Skydoc-to-Stardoc migration documentation

Fixes #146

### CODE REVIEW

**Comment by c-parsons** [comment]
I still need to manually verify the site changes get pushed OK before I merge this (I'm waiting on some permissions issues) but want to make sure things look good in principle :)

File: site/docs/skydoc_deprecation.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+### Starlark Dependencies
+
+Stardoc depends on your `bzl` file, all of its dependencies, and all of its **transitive**
+dependencies, so that it can fully evaluate your Starlark code.
+`bazel-skylib`'s `bzl_library` is your "best friend" for tracking `bzl` dependencies.
+
+For example, suppose your `mypackage/foo.bzl` file depends on `other/package/bar.bzl`, which
+depends on `third/package/baz.bzl`.
+
+```
```
Comment (brandjon) [inline site/docs/skydoc_deprecation.md]: Nit: Use a separate triple-quoted markdown block for each of these three files, to improve readability.

File: site/docs/skydoc_deprecation.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+# third/package/BUILD
+load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+
+bzl_library(
+    name = "baz",
+    srcs = ["baz.bzl"],
+)
+```
+
+Thus, each dependency should have its own `bzl_library` target defined beside it, and it
```
Comment (brandjon) [inline site/docs/skydoc_deprecation.md]: Suggestion:

Thus, each `.bzl` file should appear in the `srcs` of a `bzl_library` target defined in the same package. The `deps` of this `bzl_library` should be (only) the `bzl_library` targets corresponding to the files that are *directly* `load()`ed by the `srcs`. This structure mirrors that of other `<language>_library` rules in Bazel.

File: README.md [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
-`.bzl` file.
+Stardoc provides a Starlark rule (`stardoc`) that can be used to build documentation
+for Starlark rules in Markdown. Stardoc generates one documentation page per `.bzl` file.
 
-NOTE: `skylark_library`'s implementation has moved to
-[Skylib](https://github.com/bazelbuild/bazel-skylib#skylark_library).
+## Skydoc deprecation
 
-![A screenshot of Skydoc generated HTML documentation](https://raw.githubusercontent.com/bazelbuild/skydoc/master/skydoc-screenshot.png)
+NOTE: Stardoc is a replacement for the **deprecated** "Skydoc" documentation generator.
```
[the remark this answers is not in the document]
Comment (c-parsons) [reply README.md]: I'm not sure I do, given that we have no concrete plans. 
"We want to change the name of the repository some day but not today" doesn't sound good :) Lets update this when we have plans?

File: README.md [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
+
+See [Skydoc Deprecation](https://skydoc.bazel.build/docs/skydoc_deprecation.html) for
+details on the deprecation, and for some migration tips.
 
 ## Get Started
 
-* How to [set up Skydoc for your project](https://skydoc.bazel.build/docs/getting_started.html)
-* The Skydoc [docstring format](https://skydoc.bazel.build/docs/writing.html)
-* How to [integrate Skydoc with your build](https://skydoc.bazel.build/docs/generating.html).
+* How to [set up Stardoc for your project](https://skydoc.bazel.build/docs/getting_started.html)
```
[the remark this answers is not in the document]
Comment (c-parsons) [reply README.md]: Yes, though simply by commenting on this you've made me realize I've made an error in the link. I've updated the links.

File: site/docs/skydoc_deprecation.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        "deps" : attr.label_list(
+            doc = "Dependencies for this target.",
+        ),
+    }
+)
+```
+
+### Different Starlark Rule
+
+Stardoc uses a different Starlark rule Skydoc with different attributes.
```
[the remark this answers is not in the document]
Comment (c-parsons) [reply site/docs/skydoc_deprecation.md]: thanks. done.

File: site/docs/skydoc_deprecation.md (hunk 1 above)
Comment (c-parsons) [reply site/docs/skydoc_deprecation.md]: Done, and also pulled the file names out.

File: site/docs/skydoc_deprecation.md (hunk 2 above)
Comment (c-parsons) [reply site/docs/skydoc_deprecation.md]: Great suggestion. Thanks a bunch.

**Pull request merged**